common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.cpp

```diff
@@ -69,7 +227,161 @@ std::string sta_wlan_hal_nl80211::get_bssid() { return m_active_bssid; }
 
 bool sta_wlan_hal_nl80211::process_nl80211_event(parsed_obj_map_t &parsed_obj) { return true; }
 
-bool sta_wlan_hal_nl80211::update_status() { return false; }
+bool sta_wlan_hal_nl80211::update_status()
+{
+    ConnectionStatus connection_status;
+    if (!read_status(connection_status)) {
+        LOG(ERROR) << "Failed reading connection status for iface: " << get_iface_name();
+        return false;
+    }
+    update_status(connection_status);
+
+    return true;
+}
+
+// Add a new network and return the ID
+int sta_wlan_hal_nl80211::add_network()
+{
+    char *reply = nullptr;
+
+    std::string cmd = "ADD_NETWORK";
+
+    // Send command
+    if (!wpa_ctrl_send_msg(cmd, &reply)) {
+        LOG(ERROR) << "ADD_NETWORK failed!";
+        return false;
+    }
+
+    // Return the newly added network ID
+    return beerocks::string_utils::stoi(reply);
+}
+
+bool sta_wlan_hal_nl80211::set_network(int network_id, const std::string &param,
+                                       const std::string &value)
+{
+    const std::string cmd = "SET_NETWORK " + std::to_string(network_id) + " " + param + " " + value;
+    if (!wpa_ctrl_send_msg(cmd)) {
+        LOG(ERROR) << "SET_NETWORK failed!";
```

Comment:
this is a general comment to all APIs - perhaps it would be useful in case of error to log-print the constructed commands